### PR TITLE
Abort generation if external trigger does not match

### PR DESCRIPTION
**Pull request by lilliCao**
Implements

* Add checking if the trigger of external Increments also matches. If not throw InvalidConfigurationException with further information

@devonfw/cobigen

### CODE REVIEW

File: cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java [hunk 1]
```diff
@@ -107,6 +110,30 @@ public void testGenerationWithExternalIncrements() throws Exception {
...
+        CobiGen cobigen =
+            CobiGenFactory.create(new File(testFileRootPath + "externalIncrementsGenerationException").toURI());
+
+        // exception is thrown while getting all increments
+        assertThatThrownBy(() -> {
+            cobigen.getMatchingIncrements(input);
+        }).isInstanceOf(InvalidConfigurationException.class).hasMessageContaining(
+            "An external incrementRef to valid_increment_composition::0 is referenced from external_incrementref but its trigger does not match");
+    }
+
```
Comment (maybeec) [inline cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java:136]: Could you add also another positive test, which shows that an external template is generated?

Comment (lilliCao) [reply cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java:136]: The positive test is already written by @jdiazgon  in GenerationTest.testGenerationWithExternalIncrements

Comment (maybeec) [reply cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java:136]: ah but I thought external templates are not yet generated, right? Why could the positive test already exist?

Comment (lilliCao) [reply cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java:136]: @maybeec the external templates are always generated if the external trigger is matched. Yeah, a test exists :) Please refer to the class GenerationTest

**Comment by maybeec** [comment]
Resolves also #429

**Comment by maybeec** [comment]
Also resolves #678

**Pull request merged**